api/v1alpha1/values_types.go

```diff
@@ -0,0 +1,670 @@
+	// Controls whether various repair behaviors are enabled.
+	LabelPods           bool   `json:"labelPods,omitempty"`
+	RepairPods          bool   `json:"repairPods,omitempty"`
+	DeletePods          bool   `json:"deletePods,omitempty"`
+	BrokenPodLabelKey   string `json:"brokenPodLabelKey,omitempty"`
+	BrokenPodLabelValue string `json:"brokenPodLabelValue,omitempty"`
+	InitContainerName   string `json:"initContainerName,omitempty"`
+}
+
+type ResourceQuotas struct {
+	// Controls whether to create resource quotas or not for the CNI DaemonSet.
+	Enabled bool  `json:"enabled,omitempty"`
+	Pods    int64 `json:"pods,omitempty"`
+}
+
+// Mirrors Resources for unmarshaling.
+type Resources struct {
+	Limits   map[string]string `json:"limits,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+	Requests map[string]string `json:"requests,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+}
+
+// Mirrors ServiceAccount for unmarshaling.
+type ServiceAccount struct {
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// DefaultPodDisruptionBudgetConfig specifies the default pod disruption budget configuration.
+//
+// See https://kubernetes.io/docs/concepts/workloads/pods/disruptions/
+type DefaultPodDisruptionBudgetConfig struct {
+	// Controls whether a PodDisruptionBudget with a default minAvailable value of 1 is created for each deployment.
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// DefaultResourcesConfig specifies the default k8s resources settings for all Istio control plane components.
+type DefaultResourcesConfig struct {
+	// k8s resources settings.
+	//
+	// See https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#resource-requests-and-limits-of-pod-and-container
+	Requests *ResourcesRequestsConfig `json:"requests,omitempty"`
+}
+
+// Global Configuration for Istio components.
+type GlobalConfig struct {
+	// List of certSigners to allow "approve" action in the ClusterRole
+	CertSigners         []string `json:"certSigners,omitempty"`
+	ConfigRootNamespace string   `json:"configRootNamespace,omitempty"`
+	// Controls whether the server-side validation is enabled.
+	ConfigValidation                bool     `json:"configValidation,omitempty"`
+	DefaultConfigVisibilitySettings []string `json:"defaultConfigVisibilitySettings,omitempty"`
+	// Specifies the docker hub for Istio images.
+	Hub string `json:"hub,omitempty"`
+	// Specifies the image pull policy for the Istio images. one of Always, Never, IfNotPresent.
+	// Defaults to Always if :latest tag is specified, or IfNotPresent otherwise. Cannot be updated.
+	//
+	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
+	ImagePullPolicy  string   `json:"imagePullPolicy,omitempty"` // ImagePullPolicy             v1.PullPolicy                 `json:"imagePullPolicy,omitempty"`
```

Comment:
Why not use the k8s type here?

Reply:
I kept the same type as it is defined in [values_types.pb.go](https://github.com/istio/istio/blob/master/operator/pkg/apis/istio/v1alpha1/values_types.pb.go#L1508) (defined as a `string` in  [values_types.proto](https://github.com/istio/istio/blob/master/operator/pkg/apis/istio/v1alpha1/values_types.proto#L345))

Reply:
We decided to remove all the kubernetes related fields